extra/classes/prettyprinter/prettyprinter.factor

```diff
@@ -0,0 +1,91 @@
+! Copyright (C) 2023 Jean-Marc Lugrin.
+! See https://factorcode.org/license.txt for BSD license.
+
+! Print a class heierarchy in the listener
+! or to a file: "TESTDUMP.TXT" utf8 [ gadget hierarchy. ] with-file-writer
+
+
+USING:  classes hashtables ui.gadgets assocs kernel sequences prettyprint  vectors 
+math io formatting strings sorting accessors io.styles vocabs ;
+
+IN: classes.prettyprinter
+
+ERROR: not-a-class-error ;
+
+<PRIVATE
+
+CONSTANT: in-col 40
+
+: add-child  ( c h  -- )
+    over ! ( c h -- c h c )
+    superclass-of  ! ( c h c -- c h s )
+    swap at* ! (c h s -- c s h -- c v ? ) vector for superclass but if f ignore
+        [ swap suffix! drop ]  ! ( c v -- )
+        [ 2drop ] 
+    if 
+;
+
+: print-class-name ( c -- )
+    dup name>> swap write-object
+;
+
+
+: print-in ( c -- )
+    vocabulary>> lookup-vocab dup name>> 
+    dup ".private" tail? [ " P" ]  [ "  " ] if  write ! Mark if private
+    " IN: " write 
+    swap write-object 
+;
+
+: print-leader ( i -- )
+    ! 2 * CHAR: . <string>  write ! leader
```

Comment:
once you get it working, i suggest generally removing alternative versions of things -- unless they provide good explanatory context, or you use the lines to swap between two implementations or something